deploy-tool/main.go

```diff
@@ -132,26 +132,85 @@ func generateAndDeployKeyPair(
 	return p256Key, nil
 }
 
-func manifestExists(fqdn, dsp string) bool {
-	// Remove the locality name from the FQDN
-	path := fmt.Sprintf("https://%s/%s-manifest.json", fqdn, dsp)
+// globalManifestExists checks if a global data share processor manifest exists
+// relative to the manifest base URL provided, returning (true, nil) if so,
+// (false, nil) if not, and (false, error) if it could be determined
+func globalManifestExists(manifestBaseURL string) (bool, error) {
+	url := fmt.Sprintf("https://%s/global-manifest.json", manifestBaseURL)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return false, fmt.Errorf("error fetching manifest %s: %w", url, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == 403 || resp.StatusCode == 404 {
```

Comment:
nit: We could probably abstract away the http call and error handling into its own function. I believe we're doing these generic if checks in multiple places now.

Reply:
J.C. made a similar observation, and my answer is that all of this gets refactored away in #763, which is context I should have provided in this PR.